ansible/dualtor/nic_simulator/nic_simulator_grpc_service.proto

```diff
@@ -1,11 +1,15 @@
 syntax = "proto3";
 
-service DualTorService {
```

Comment:
Suggest using this as is

service DualToRActive {
  rpc QueryAdminForwardingPortState(AdminRequest) returns (AdminReply) {}
  rpc SetAdminForwardingPortState(AdminRequest) returns (AdminReply) {}
  rpc QueryOperationPortState(OperationRequest) returns (OperationReply) {}
  rpc QueryLinkState(LinkStateRequest) returns (LinkStateReply) {}
  rpc QueryServerVersion(ServerVersionRequest) returns (ServerVersionReply) {}
}
message AdminRequest {
  repeated int32 portid = 1;
 repeated bool state = 2;
}
message AdminReply {
  repeated int32 portid = 1;
  repeated bool state = 2;
}
message OperationRequest {
  repeated int32 portid = 1;
}
message OperationReply {
  repeated int32 portid = 1;
  repeated bool state = 2;
}
message LinkStateRequest {
  repeated int32 portid = 1;
}
message LinkStateReply {
  repeated int32 portid = 1;
  repeated bool state = 2;
}
message ServerVersionRequest {
  string version = 1;
}
message ServerVersionReply {
  string version = 1;
}

Reply:
@vdahiya12, hi, updated with changes to `QueryAdminForwardingPortState` and `SetAdminForwardingPortState`.